### DISCUSSION TITLE
Making Ybus dense/filling in entries

### DISCUSSION THREAD

**Discussion by bergkvist** [e2nIEE/pandapower#1161] [General]
The Ybus (`net._ppc['internal']['Ybus']`) is a sparse matrix where only the diagonal and a few other elements seem to be non-zero. (after `pp.runpp(net)`).

Is there an easy way of making this matrix more "dense"?

In particular I'm interested in using dSbus_dV to find an approximation for how power injection at certain buses will affect the voltage for buses that are located somewhere else in the grid.
```py
from pandapower.pf.dSbus_dV_numba import dSbus_dV

Ybus = net['_ppc']['internal']['Ybus']
V = net['_ppc']['internal']['V']
dS_dVm, dS_dVa = dSbus_dV(Ybus, V)
```

Let's say I want to inject power into "bus 31" - and I wonder how the voltage at "bus 14" and "bus 15" will be affected by this (approximately)

**Comment by wangzhenassd** [comment]
Hi, I think if you want to find out the sensitivity dVm_dp, dVm_dq you need the inverse of the Jacobian matrix from NR Powerflow

**Reply by bergkvist** [reply]
The jacobian matrix is sparse, so it doesn't tell you how buses far away from each other affect each other, or does it?

**Reply by wangzhenassd** [reply]
[A Sensitivity Analysis Toolkit for the Simplification of MV Distribution Network Voltage Management](https://ieeexplore.ieee.org/document/6729136) I think this paper explains well how the inverse of J matrix represents the voltage sensitivity
